examples/stm32h7-example/README.md

```diff
@@ -0,0 +1,55 @@
+# STM32H7 example
+This STM32 UART example is created for the `STM32H750IBK6 MCU`, placed on a `Electrosmith Daisy Seed` development board. But below I have described the process, so it should be recreate-able for every STM32 MCU.
+
+# STM32CubeMX general settings
+This example is using UART interrupts for each received character, and has re-routed (and buffered) _write function for easy prints to the CLI. All written in C.
+
+I have created it using the STM32 HAL, and using the STM32CubeMX configurator (.ioc file), but the process should be more/less the same for LL drivers & using custom cmake files.
+
+This example uses the STM32CubeMX option: "Generate peripheral initialization as pair of `.c/.h` files per peripheral", since I personally like that option enabled. It is found under "Project Manager/Code Generator" in `STM32CubeMX`.
+
+I've also split the initialization and command binding up into multiple files, for easier re-use in other projects.
+
+# Steps to recreate from IOC file
+**Step 0.**<br>
+Import project with using .ioc file to your own workspace (can be done by right clicking in `STM32CubeIDE->import->Import an Existing STM32CubeMX Configiration File (.ioc)`, or can be done directly from standalone `STM32CubeMX`). If using this .ioc file, you could skip step 1, step 2 and step 3 (these steps should already be set in the delivered .ioc file, but it's always a good idea to double check).
+
+**Step 1.**<br>
+Enable U(S)ART in asynchronous mode. This example uses USART1, 115200 Bits/s, 8 bit words, no parity and 1 stop bit. The H7 family has quite a lot of advanced USART features. Disable at least: `Auto Baudrate`, `Overrun` and `DMA on RX Error`.
+
+**Step 2.**<br>
+Enable USARTx Global Interrupt in the NVIC. I personally like to enable interrupt `select for init sequence order` to make sure this is the first global interrupt to be handled.
+
+**Step 3.**<br>
+Configure your clock correctly. The board I used for this example (Electrosmith Daisy Seed), has an 16 MHz external resonator. So in RCC, I enabled HSE to: `Crystal/Ceramic resonator` with the default settings.
+Then in clock configurator I set the input frequency to 16MHz, and solved my clock issues to get an USART1 frequency of 32MHz.
+
+**Step 4.**<br>
+Include the following `.c/.h` file combos into their corresponding source/header directories:
+* `embedded_cli.c/.h` (can be found in the root of this repository, in the folder `lib`)<br>
```

Comment:
Probably should replace with single header instruction. I have plans to split `embedded_cli.c` and `embedded_cli.h` into multiple files for easier development while still distributing single header version for easier usage.